src/sage/combinat/designs/orthogonal_arrays.py

```diff
@@ -1543,8 +1553,42 @@ def OA_relabel(OA,k,n,blocks=tuple(),matrix=None):
     if matrix:
         OA = [[matrix[i][j] if j is not None else None for i,j in enumerate(R)] for R in OA]
 
+    if symbol_list:
+        result=[]
+        the_map = lambda x: symbol_list[x]
+        for row in OA:
+            result.append(list(map(the_map,row)))
+        OA = result
     return OA
 
+def OA_standard_label(OA):
+    r"""
+    Return the inputted OA with entries relabelled as integers [0,...,n-1].
+
+    INPUT:
+
+    - ``OA`` -- a list of lists with symbols as entries that are not
+      necessarily integers.
+
+    EXAMPLES::
+
+        sage: from sage.combinat.designs.orthogonal_arrays import OA_standard_label
+        sage: C = [['a', 'a', 'a', 'b'],
+        ....:      ['a', 'a', 'b', 'a'],
+        ....:      ['a', 'b', 'a', 'a'],
+        ....:      ['b', 'a', 'a', 'a'],
+        ....:      ['b', 'b', 'b', 'b']]
+        sage: OA_standard_label(C)
+        [[0, 0, 0, 1], [0, 0, 1, 0], [0, 1, 0, 0], [1, 0, 0, 0], [1, 1, 1, 1]]
+
+    """
+    symbol_list = sorted({x for l in OA for x in l})
+    result = []
+    the_map = lambda x: symbol_list.index(x)
```

Comment:
This takes quadratic time for no good reason. You should rather use a dictionary
```
mapping = {symbol : index for index, symbol in enumerate(symbol_list)}
```
However, that would fail for non hashable symbols. Nevertheless you also assumed the symbols to be hashable in your first line.

Reply:
+1 on this change; almost everything people will use will be hashable (or at least should be and just needs the immutable version).

Reply:
I improved both functions that use this line with your suggestion